samples/openthread/ncp/README.rst

```diff
@@ -0,0 +1,264 @@
+.. _ot_ncp_sample:
+
+Thread: NCP
+###########
+
+The Thread NCP sample demonstrates the usage of OpenThread Network Co-Processor architecture inside the Zephyr shell.
+
+The sample is based on Zephyr's :ref:`zephyr:ncp-sample` sample.
+However, it customizes Zephyr's sample to the NCS requirements (for example, by increasing the stack size dedicated for the user application), and also extends it with several features:
+
+* Increased mbedTLS heap size.
+* Lowered main stack size to increase user application space
+* No obsolete configuration options.
+
+Overview
+********
+
+The sample demonstrates using an NCP target on the MCU to communicate with Userspace WPAN Network Daemon (wpantund) on Unix-like operating system.
+According to the NCP architecture, the MCU part needs to cooperate with user higher layer process to establish the complete full stack application.
+The sample shows how to set connection between NCP and wpantund.
+
+This sample has Commissioner, Joiner, and Border Router roles enabled by default.
+Additionally, it comes with the following functionalities enabled:
+
+* :option:`CONFIG_OPENTHREAD_BORDER_AGENT` - Enables Border Agent support.
+* :option:`CONFIG_OPENTHREAD_DHCP6_SERVER` - Enables DHCPv6 server capability.
+
+Requirements
+************
+
+The sample supports the following development kits for testing the network status:
+
+.. include:: /includes/boardname_tables/sample_boardnames.txt
+   :start-after: set8_start
+   :end-before: set8_end
+
+To test the sample, you need at least one development kit NCP board.
+Additional NCP boards can be used for the :ref:`optional testing of network joining <ot_ncp_sample_testing_more_boards>`.
+
+Moreover, the sample requires a Userspace higher layer process running on user's device in order to communicate with the MCU NCP part.
+This sample uses `wpantund`_ as reference.
+
+User interface
+**************
+
+All the interactions with the application are handled using serial communication.
+
+For the interaction with the application, this sample uses `wpantund`_ process with its commands.
```

Comment:
You control wpantund with dbus, 'wpanctl' is the tool which provides command interface.

Reply:
Done